website/docs/r/service_acl_entries_v1.html.markdown

```diff
@@ -0,0 +1,123 @@
+---
+layout: "fastly"
+page_title: "Fastly: service_acl_entries_v1"
+sidebar_current: "docs-fastly-resource-service-v1"
+description: |-
+  Defines a set of Fastly ACL entries that can be used to populate a service ACL. 
+---
+
+# fastly_service_acl_entries_v1
+
+Defines a set of Fastly ACL entries that can be used to populate a service ACL.  This resource will populate an ACL with the entries and will track their state.
+
+~> **Warning:** Terraform will take precedence over any changes you make in the UI or API. Such changes are likely to be reversed if you run terraform again.  
+
+If Terraform is being used to populate the initial content of an ACL which you intend to manage via API or UI, then the lifecycle `ignore_changes` field can be used with the resource.  An example of this configuration is provided below.    
+
+
+## Example Usage
+
+Basic usage:
+
+```hcl
+variable "myacl_name" {
+	type = string
+	default = "My ACL"
+}
+
+resource "fastly_service_v1" "myservice" {
+  name = "demofastly"
+
+  domain {
+      name = "demo.notexample.com"
+      comment = "demo"
+  }
+
+  backend {
+      address = "demo.notexample.com.s3-website-us-west-2.amazonaws.com"
+      name = "AWS S3 hosting"
+      port = 80
+    }
+
+  acl {
+	name = var.myacl_name
+  }
+
+  force_destroy = true
+}
+
+resource "fastly_service_acl_entries_v1" "entries" {
+  service_id = fastly_service_v1.myservice.id
+  acl_id = {for d in fastly_service_v1.myservice.acl : d.name => d.acl_id}[var.myacl_name]
+  entry {
+    ip = "127.0.0.1"
+    subnet = "24"
+    negated = false
+    comment = "ALC Entry 1"
+  }
```

Comment:
Please add another entry to the example. It might help users to understand the syntax of an ACL with multiple entries.

I would label this example as "Basic usage". Please add another example (Complex object usage) which applies the new "dynamic nested block" syntax for an ACL with multiple entries and with static fields. Here is a complete example which you can use if you like:

```
locals {
  acl_name = "my_acl"
  acl_entries = [
    {
      ip      = "1.2.3.4"
      comment = "acl_entry_1"
    },
    {
      ip      = "1.2.3.5"
      comment = "acl_entry_2"
    },
    {
      ip      = "1.2.3.6"
      comment = "acl_entry_3"
    }
  ]
}

resource "fastly_service_v1" "myservice" {
  name = "demofastly"

  domain {
    name    = "demo.notexample.com"
    comment = "demo"
  }

  backend {
    address = "1.2.3.4"
    name    = "localhost"
    port    = 80
  }

  acl {
    name = local.acl_name
  }

  force_destroy = true
}

resource "fastly_service_acl_entries_v1" "entries" {
  service_id = fastly_service_v1.myservice.id
  acl_id     = { for d in fastly_service_v1.myservice.acl : d.name => d.acl_id }[local.acl_name]
  dynamic "entry" {
    for_each = [for e in local.acl_entries : {
      ip      = e.ip
      comment = e.comment
    }]

    content {
      ip      = entry.value.ip
      subnet  = 22
      comment = entry.value.comment
      negated = false
    }
  }
}
```

Reply:
The example was executed and placed into the documentation.